### ISSUE TITLE
SAXParserFactoryImpl not found

### ISSUE THREAD

**Issue by NoelOC** [KevM/tikaondotnet#21]
Hi,

I'm getting the following errors when trying an extraction: 
Provider com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl not found
Provider com.sun.org.apache.xalan.internal.xsltc.trax.TransformerFactoryImpl not found

I have manage to workaround the issue by adding the following before calling a new TextExtractor()
com.sun.org.apache.xerces.@internal.jaxp.SAXParserFactoryImpl s = new com.sun.org.apache.xerces.@internal.jaxp.SAXParserFactoryImpl();
                        com.sun.org.apache.xalan.@internal.xsltc.trax.TransformerFactoryImpl t = new com.sun.org.apache.xalan.@internal.xsltc.trax.TransformerFactoryImpl();

This forces the files to be loaded and all works fine.

I have added the appSettings in the app.config as mentioned in your blog post but it still throws this error.

It's no big deal, I had run it through over 2k documents and it extracted the text perfectly.
I was just wondering was I missing something else in the setup process.

Thank you for making this. It's been a joy to use. It makes extracting text from documents very easy on .net.

**Comment by KevM** [comment]
Glad to hear you got it working. I have not run into that problem before. Why kind of documents are you extracting when you ran into that problem? It sounds a bit like a Java class loader problem.

**Comment by NoelOC** [comment]
It happened on every document so I also think it's a Java class loader problem.

It's a web application running on Azure so the project is broken up like this:
MyApp.Shared
MyApp.Service
MyApp.Web 
etc

I run the TextExtractor in the Service project above, which is also where I've installed your nuget package in the application.
I added the appSettings like to app.config of the MyApp.Service and also the web.config of MyApp.Web

Not sure if any of that makes sense.
Like I said, it works fine but it feels a little dirty having to load the classes like that every time.

Thanks.

**Comment by jrotello** [comment]
I've noticed in the shared library scenario that the Copy Local of IKVM is a little flaky. For example, in my project only 20 of the 25 IKVM assemblies get copied local to MyApp.Service or MyApp.Web. If that is the case for you, you have two choices that I know of:

1. Explicitly reference the IKVM package in MyApp.Service and MyApp.Web.
2. Manually copy the IKVM assemblies to MyApp.Service and MyApp.Web as a post build step or similar.

**Comment by NoelOC** [comment]
Thanks for the suggestion.

I've checked the project and Copy Local is set to 'true' for all the IKVM files in the project.

Your comment made me think.
I was calling the MyApp.Service from a worker role that reads off a queue.

I've added the Tikaondotnet nuget package to the worker role.
Added the appSettings in the app.config of the worker role and it worked without the 2 lines I've added above.

Thank you for your help. It was the mention of 'Shared library' that trigger the change.
